Felienne/hedy
# Highlight firsts levels

In order to solve the problem of syntactic over-coloring, it is necessary to replace the current system by a new system: a more complex automaton.

This pull request does not solve the problem of over-coloring, but prepares the solution, by starting the replacement of the system of the first levels (the old system was an exhaustive system). This change of system corrects 2 bugs that were present.


For readability reasons, some transitions are not displayed: 
- From each of the states there is a transition for comments and one for the line break, which sets the state to start.
- From each state S there are 2 transitions for invalid characters ( _ and _?_ ) which do not change the state (we return to the start state).

Here is a visual description of the 3 syntax highlighting automata: 

## Level 1 :
![image](https://user-images.githubusercontent.com/48509466/177775098-4e3c4763-bb67-4150-977e-060a2dfb6c1b.png)
Description of the automaton:
- The commands ask, echo, print and forward take us from the start of line state to the "Value" state. This state does not recognize anything.
- The turn command takes us to a state where we recognize only the direction commands
- The command color takes us to a state where we recognize color names. 

## Level 2 :
![image](https://user-images.githubusercontent.com/48509466/177775885-847e3f7e-75d5-4972-a553-83a6f9e8ea95.png)
Description of the automaton:
for all commands, the state after will not recognize anything (left and right don't exist anymore, and the command color doesn't exist at level 2)

## Level 3 :
![image](https://user-images.githubusercontent.com/48509466/177777747-54e4419d-9e33-437b-9176-dade8c90fd1d.png)
Description of the automaton:
for all commands, the state after will not recognize anything (left and right don't exist anymore, and the command color doesn't exist at level 2)

### Tests

When modifying the syntactic highlighting of level 3, the highlighting of commas, and the recognition of blanks was added to level 3, and it was therefore necessary to modify some tests:

Before :
![image](https://user-images.githubusercontent.com/48509466/177773025-35e58614-6974-43a4-b8b6-97564b838b21.png)
![image](https://user-images.githubusercontent.com/48509466/177772621-e4991a7a-68d3-4db1-8b0e-3d9e8e7b3b4e.png)

Now :
![image](https://user-images.githubusercontent.com/48509466/177773096-651dd4ed-0a76-4ce3-a384-3bfc41776e5c.png)
![image](https://user-images.githubusercontent.com/48509466/177772525-2f4bfaeb-1897-46e1-baef-d04404bd0705.png)

# How to test
Check that it doesn't break anything for level 3, both from the front-end and syntactic highlighting point of view.
